demisto_sdk/commands/common/tools.py

```diff
@@ -780,11 +792,38 @@ def _read_file(file_path: Path) -> str:
 
     except UnicodeDecodeError:
         try:
-            # guesses the original encoding
-            return UnicodeDammit(file_path.read_bytes()).unicode_markup
+            logger.debug(f"Could not read file '{file_path}' using UTF-8 encoding. Trying to auto-detect encoding...")
+            return UnicodeDammit(file_path.read_bytes()).unicode_markup  # Guess the original encoding
+
+        except UnicodeDecodeError:
+            logger.debug(f"Could not auto-detect encoding for file '{file_path}'.")
+            raise
+
+
+def create_stringio_object(file_data: bytes | str) -> StringIO:
```

Comment:
How about we create our own StringIO class (inherit from the real one, wrap its constructor), and ban using the real StringIO? (see last lines of `pyproject.toml`)

Reply:
Why? What is the benefit of it?
Seems to me like making stuff complex for no reason / benefit.